aruco_tags/aruco_tag_detection/src/aruco_marker_visualization.cpp

```diff
@@ -0,0 +1,63 @@
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_components/register_node_macro.hpp>
+#include <stsl_interfaces/msg/tag_array.hpp>
+#include <visualization_msgs/msg/marker.hpp>
+#include <visualization_msgs/msg/marker_array.hpp>
+
+using namespace std::chrono_literals;
+
+namespace aruco_tag_detection
+{
+class ArucoVisualization : public rclcpp::Node
+{
+public:
+  ArucoVisualization(const rclcpp::NodeOptions & options)
+  : rclcpp::Node("aruco_tag_visualization", options)
+  {
+    marker_pub_ = create_publisher<visualization_msgs::msg::MarkerArray>("true_tags", 1);
+    tag_sub_ = this->create_subscription<stsl_interfaces::msg::TagArray>("/aruco_tag_detector/tags",
+               10, std::bind(&ArucoVisualization::tagCallback, this, std::placeholders::_1));
+  }
+private:
+  rclcpp::Publisher<visualization_msgs::msg::MarkerArray>::SharedPtr marker_pub_;
+  rclcpp::Subscription<stsl_interfaces::msg::TagArray>::SharedPtr tag_sub_;
+
+  void tagCallback(const stsl_interfaces::msg::TagArray::SharedPtr tag_array_msg) {
+    visualization_msgs::msg::MarkerArray marker_array_msg;
+
+    for(const stsl_interfaces::msg::Tag & tag : tag_array_msg->tags)
+    {
+      visualization_msgs::msg::Marker marker_msg;
+      marker_msg.id = tag.id;
```

Comment:
Does it matter that this node and the student code both use the same marker IDs? Do they not overwrite each other's markers?

Reply:
I think this might be the reason you're seeing that blipping? Conflicting IDs from two different sources.